extensions/amp-analytics/0.1/test/test-visibility-model.js

```diff
@@ -1121,4 +1122,111 @@ describes.sandboxed('VisibilityModel', {}, () => {
       expect(spy).to.be.calledTwice;
     });
   });
+
+  describe('reportWhen', () => {
+    beforeEach(() => {
+      sandbox.restore(); // Restore timers.
+    });
+
+    // Yields the thread to allow internal promises to complete.
+    const yieldThread =
+      async() => new Promise(resolve => setTimeout(resolve, 1));
```

Comment:
nit: would `setTimeout(,0)` work? You could do `yield macroTask()` for that.

Reply:
Thanks for the reference!

Reply:
No longer applicable